src/main/java/com/softwire/todos/TodoCheckerMain.java

```diff
@@ -50,15 +54,14 @@
             required = false)
     public String jiraUrl = "https://jira.softwire.com/jira/";
 
-    @Option(name = "--jira-project-key",
-            usage = "The project key for JIRA, e.g. AAA, INTRO, BBC, PROJECTX, etc",
-            required = true)
-    public String jiraProjectKey;
-
-    @Option(name = "--jira-project-key-in-todo-regex",
-            usage = "A regex for project key as used in a todo, which doesn't need to match the jira key",
-            required = true)
-    public String jiraProjectKeyInTodoRegex;
+    @Option(name = "--jira-project",
+            usage = "The project key for JIRA, e.g. AAA, INTRO, PROJECTX, etc.  Pass this flag multiple times for " +
+                    "multiple projects.  If you want to use a regex for the project key as used in a todo, for " +
+                    "example if your JIRA project is FOO but in TODOs it appears as BAR, then pass " +
+                    "--jira-project FOO=BAR",
+            required = true,
+            handler = JiraProjectOptionHandler.class)
+    public List<JiraProject> jiraProjects;

```

Comment:
Thanks, this looks good to me.

I would say:

 1. Please consider renaming the option back to "jira-project-key". The value here is the alphanumeric *key* for the Jira Project, which is a distinct concept from the Jira project itself. I think that the term "jira project" is ambiguous by itself and will lead to users needing to guess (or look up in the docs) what property of the jira project is expected (e.g. its URL, its name, its key?).

 2. Please clarify the example. The use case which motivated this option is a project with a long JIRA key which includes the customer name, where TODOs in the code use an abbreviation. How about:

```
If you need to use a regex other than the project key when looking for the card key
in a todo comment, then pass it here with an "=". For example if your JIRA project key
is something long like COMPANY-DEPT-FOO but your team writes TODOs like
"TODO:FOO-123", then pass "--jira-project COMPANY-DEPT-FOO=FOO",
```

Reply:
Thanks. I agree both of those are clearer, will fix.